CMakeLists.txt

```diff
@@ -63,14 +63,22 @@ endif()
 add_subdirectory(src)
 
 # add usage example
-add_subdirectory(example)
+option(BUILD_EXAMPLES "Build usage examples" ON)
+if(BUILD_EXAMPLES)
+    add_subdirectory(example)
+endif()
 
 # generate project documentation
-add_subdirectory(docs)
+option(BUILD_DOCS "Generate docs" ON)
+if(BUILD_DOCS)
+    add_subdirectory(docs)
+endif()
```

Comment:
This is already provided by the `./docs/CMakeLists.txt` file.

Reply:
True, I didn't noticed.